Add rfactor patterns for NaN-propagating min/max

alexreinking opened the pull request:
These are required to write schedules for PyTorch pipelines, whose operators use these semantics for min/max.

These rules were checked by Z3 manually.

abadams commented on src/AssociativeOpsTable.cpp:
```diff
@@ -249,6 +249,29 @@ void populate_ops_table_single_uint32_select(const vector<Type> &types, vector<A
     table.emplace_back(select(x0 < -y0, y0, tmax_0), zero_0, true);          // Saturating add
 }
 
+Expr is_nan_not_strict(Expr x) {
```
is_nan without strict_float can legally simplify to false, so this code threw me at first. But I realize what you're writing here is not an expression with a value - it's a pattern you're trying to pattern-match an expression against, and you don't want the strict_float to be part of the pattern because you don't want to strip it. If that's correct, please add a clarifying comment.

alexreinking replied:
Done

alexreinking commented:
I've rebased this on main now that #8597 has merged. I added the requested comment and added a test for both the bitwise-or and logical-or forms. These are designed to be correct on their own and will break if #8597 is reverted without updating the rfactor logic.
